boot/zephyr/flash_map_extended.c

```diff
@@ -140,3 +140,24 @@ __weak uint8_t flash_area_erased_val(const struct flash_area *fap)
     (void)fap;
     return ERASED_VAL;
 }
+
+int flash_area_get_sector(const struct flash_area *fap, off_t off,
+                          struct flash_sector *fsp)
+{
+    struct flash_pages_info fpi;
+    int rc;
+
+    if (off >= fap->fa_size) {
+        return -ERANGE;
+    }
+
+    rc = flash_get_page_info_by_offs(fap->fa_dev, fap->fa_off + off,
+            &fpi);
+
+    if (rc == 0) {
+        fsp->fs_off = fpi.start_offset - fap->fa_off;
```

Comment:
Trying to work back and check this is correct, should this be take away? If start offset is e.g. 0x0, and desired offset is 0x200, the offset would be 0x0 - 0x200 which would underflow to 0xFFFFFE00

Reply:
Let me think.
The `flash_get_page_info_by_offs` returns page info in context of device, not flash area. So the `fpi.start_offset` will be greater or equal `fap->fa_off`, unless area is not aligned to flash page.
So even if `off` is set to 0, the `fsp->fs_off` should be 0 or next page start offset, unless area is not aligned to flash page.
I should return error here in case when `fpi.start_offset` is less than `fap->fa_off` and replace type of `off` as size_t to prevent negative offsets.
@nordicjm  That is what you want here?

Reply:
@nordicjm Do you want me to do the error check for the case with bad flash page alignment, where page starts before area offset?

Reply:
Don't think there's a need for now